test/features/cxx_qt_stdexec.cpp

```diff
@@ -0,0 +1,103 @@
+#include <qthread/qthread.h>
+#include <stdexec/execution.hpp>
+#include <stdio.h>
+
+#include "argparsing.h"
+
+namespace qthreads {
+struct qthreads_scheduler {
+  constexpr qthreads_scheduler() = default;
+
+  bool operator==(qthreads_scheduler const &rhs) const noexcept { return true; }
+  bool operator!=(qthreads_scheduler const &rhs) const noexcept {
+    return !(*this == rhs);
+  }
+
+  template <typename Receiver>
+  struct operation_state {
+    [[no_unique_address]] std::decay_t<Receiver> receiver;
+
+    template <typename Receiver_>
+    operation_state(Receiver_ &&receiver):
+      receiver(std::forward<Receiver_>(receiver)) {}
+
+    operation_state(operation_state &&) = delete;
+    operation_state(operation_state const &) = delete;
+    operation_state &operator=(operation_state &&) = delete;
+    operation_state &operator=(operation_state const &) = delete;
+
+    static aligned_t task(void *arg) noexcept {
+      auto *os = static_cast<operation_state *>(arg);
+      printf("Hello from qthreads in task! id = %i\n", qthread_id());
+      stdexec::set_value(std::move(os->receiver));
+      return 0;
+    }
+
+    friend void tag_invoke(stdexec::start_t, operation_state &os) noexcept {
+      aligned_t ret = 0;
+      int r = qthread_fork(&task, &os, &ret);
+      qthread_readFF(NULL, &ret);
```

Comment:
Is this the right place to put the `readFF` call? I think this is roughly the same as running `task` synchronously since the calling thread waits on the `readFF` call.

Reply:
I was wondering the same.

Just to clarify, the `sync_wait` sender consumer sets up a condition variable that it starts waiting for. The condition variable is signaled by the `set_value` call that is called inside `task`. So synchronization is taken care of without the `qthread_readFF`, but this only guarantees synchronization until user-supplied callables have finished running. The qthread task obviously hasn't fully finished running when `sync_wait` returns, and I had assumed this would be ok to do.

It is possible to customize `sync_wait` and/or provide a fiber-specific `sync_wait` that yields the fiber instead of the underlying OS-thread. In this particular example it doesn't matter since `sync_wait` is called on `main`, and in general you want to avoid calling `sync_wait` except at the top level. I think if one sets up a customized `sync_wait` that one could use qthreads' own synchronization mechanisms, like `readFF`.